src/middleware/auth.js

```diff
@@ -9,9 +10,12 @@ export const protect = asyncHandler(async (req, res) => {
       : null
     : null;
   if (!token) return makeResponse({ res, status: 403, message: 'Unauthorized' });
+  const isBackListedToken = isTokenBackListed(token);
+  if (isBackListedToken) return makeResponse({ res, status: 403, message: 'Unauthorized' });
   const decodedUser = decodeJwtToken(token).data;
   const user = decodedUser ? await getOneUser({ _id: decodedUser._id }, false) : null;
   if (!user) return makeResponse({ res, status: 403, message: 'Unauthorized' });
+  user.token = token;
```

Comment:
Shall we add this to another property of the user request like `req.user_token` maybe instead of the user as it is, cuz this user will be circulated and returned as it is in the get current user endpoint, we'll be sending back redundant info over there.